srml/babe/src/lib.rs

```diff
@@ -179,6 +172,33 @@ impl<T: Trait> RandomnessBeacon for Module<T> {
 	}
 }
 
+/// The data in a BABE pre-runtime digest
+type BabePreRuntimeData = (
+	[u8; VRF_OUTPUT_LENGTH],
+	[u8; VRF_PROOF_LENGTH],
+	BabeKey,
+	u64,
+);
+
+
+pub fn to_babe_pre_digest((engine, mut data): (ConsensusEngineId, &[u8])) -> Option<BabePreRuntimeData> {
+	if engine == BABE_ENGINE_ID { Decode::decode(&mut data) } else { None }
+}
+
+/// A BABE public key
+pub type BabeKey = [u8; PUBLIC_KEY_LENGTH];
+
+impl<T: Trait> FindAuthor<BabeKey> for Module<T> {
```

Comment:
Since we'll want to compose this with`session::FindAccountFromAuthorIndex`, it's good to also have an implementation of `FindAuthor<u32>`.